llvm/lib/Analysis/LoopAccessAnalysis.cpp

```diff
@@ -3004,7 +3004,7 @@ void LoopAccessInfo::collectStridedAccess(Value *MemAccess) {
   // of various possible stride specializations, considering the alternatives
   // of using gather/scatters (if available).
 
-  const SCEV *BETakenCount = PSE->getBackedgeTakenCount();
+  const SCEV *BETakenCount = PSE->getSymbolicMaxBackedgeTakenCount();
```

Comment:
Not related to your change - but this whole block of code is just weird.  This is basically proving a more precise trip count, why is it in LAA at all?  Wouldn't simply early exiting on small BTC loops be sufficient?

Reply:
Yeah, let me take a look into this separately.